Can't install Wazuh agents in OpenBSD platforms

|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.7.x-3.9.0 | Agent | Sources | OpenBSD |

Hi team,

As reported in https://groups.google.com/d/msg/wazuh/F5Zoxd-kRPM/LgVr2lKUAgAJ, there's an error when building the agents in FreeBSD. Here you can see the output:

```
libwazuhext.so: warning: strcpy() is almost always misused, please use 
strlcpy() 
libwazuh.a(agent_op.o): In function `w_remove_multigroup': 
agent_op.c:(.text+0x102f): warning: sprintf() is often misused, please 
use snprintf() 
libwazuhext.so: warning: strcat() is almost always misused, please use 
strlcat() 
     CC syscheckd/win_whodata.o 
     CC syscheckd/run_realtime.o 
     CC syscheckd/win-registry.o 
     CC syscheckd/syscheck_audit.o 
     CC syscheckd/run_check.o 
     CC syscheckd/seechanges.o 
     CC syscheckd/config.o 
     CC syscheckd/syscheck.o 
     CC syscheckd/syscom.o 
     CC syscheckd/create_db.o 
     CC rootcheck/check_rc_trojans.o 
     CC rootcheck/check_rc_pids.o 
     CC rootcheck/check_rc_policy.o 
     CC rootcheck/run_rk_check.o 
     CC rootcheck/check_rc_sys.o 
     CC rootcheck/check_rc_dev.o 
     CC rootcheck/check_rc_readproc.o 
     CC rootcheck/win-common.o 
     CC rootcheck/os_string.o 
     CC rootcheck/check_open_ports.o 
     CC rootcheck/check_rc_if.o 
In file included from rootcheck/check_rc_if.c:27: 
In file included from rootcheck/rootcheck.h:13: 
./headers/list_op.h:32:5: error: unknown type name 'pthread_rwlock_t' 
     pthread_rwlock_t wr_mutex; 
     ^ 
./headers/list_op.h:33:5: error: unknown type name 'pthread_mutex_t' 
     pthread_mutex_t mutex; 
     ^ 
2 errors generated. 
gmake[1]: *** [Makefile:1157: rootcheck/check_rc_if.o] Error 1 
gmake[1]: Leaving directory '/tmp/gg/wazuh-3.7.0/src' 
gmake: *** [Makefile:559: agent] Error 2 

  Error 0x5. 
  Building error. Unable to finish the installation. 
```

Regards.

--- a/src/Makefile
+++ b/src/Makefile
@@ -142,7 +142,7 @@ ifeq (${uname_S},NetBSD)
 		OSSEC_LDFLAGS+='-Wl,-rpath,$$ORIGIN/../lib'
 else
 ifeq (${uname_S},OpenBSD)
-#		DEFINES+=-DOpenBSD
+		DEFINES+=-DOpenBSD
 		OSSEC_CFLAGS+=-pthread
 		OSSEC_LDFLAGS+=-pthread
 		OSSEC_LDFLAGS+=-L/usr/local/lib

--- a/src/Makefile
+++ b/src/Makefile
@@ -146,7 +146,7 @@ ifeq (${uname_S},OpenBSD)
 		OSSEC_CFLAGS+=-pthread
 		OSSEC_LDFLAGS+=-pthread
 		OSSEC_LDFLAGS+=-L/usr/local/lib
-		OSSEC_LDFLAGS+='-Wl,-rpath,$$ORIGIN/../lib'
+		OSSEC_LDFLAGS+=-Wl,-zorigin '-Wl,-rpath,$$ORIGIN/../lib'
 else
 ifeq (${uname_S},HP-UX)
 		DEFINES+=-DHPUX
@@ -763,7 +763,11 @@ $(EXTERNAL_CURL)Makefile:
 	cd $(EXTERNAL_CURL) && ./configure --with-darwinssl --disable-ldap --without-libidn2
 else
 $(EXTERNAL_CURL)Makefile: $(OPENSSL_LIB)
+ifeq (${uname_S},Linux)
 	cd $(EXTERNAL_CURL) && CPPFLAGS="-fPIC -I${ROUTE_PATH}/${EXTERNAL_OPENSSL}include" LDFLAGS="-L${ROUTE_PATH}/${EXTERNAL_OPENSSL}" LIBS="-ldl -lpthread" ./configure --disable-ldap --without-libidn2
+else
+	cd $(EXTERNAL_CURL) && CPPFLAGS="-fPIC -I${ROUTE_PATH}/${EXTERNAL_OPENSSL}include" LDFLAGS="-L${ROUTE_PATH}/${EXTERNAL_OPENSSL}" LIBS="-lpthread" ./configure --disable-ldap --without-libidn2
+endif
 endif
 
 

--- a/src/analysisd/analysisd.c
+++ b/src/analysisd/analysisd.c
@@ -16,8 +16,11 @@
 #define ARGV0 "ossec-analysisd"
 #endif
 
-#include <time.h>
 #include "shared.h"
+#include <time.h>
+#if defined(__MACH__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#include <sys/sysctl.h>
+#endif
 #include "alerts/alerts.h"
 #include "alerts/getloglocation.h"
 #include "os_execd/execd.h"

--- a/src/headers/shared.h
+++ b/src/headers/shared.h
@@ -127,8 +127,10 @@ typedef uint8_t u_int8_t;
 
 #endif /* SOLARIS */
 
-#if defined HPUX
+#if defined(HPUX) || defined(DOpenBSD)
 #include <limits.h>
+typedef uint64_t u_int64_t;
+typedef int int32_t;
 typedef uint32_t u_int32_t;
 typedef uint16_t u_int16_t;
 typedef uint8_t u_int8_t;

--- a/src/headers/shared.h
+++ b/src/headers/shared.h
@@ -71,6 +71,7 @@
 #include <ctype.h>
 #include <signal.h>
 #include <stdbool.h>
+#include <pthread.h>
 
 /* The mingw32 builder used by travis.ci can't find glob.h
  * Yet glob must work on actual win32.

--- a/src/rootcheck/check_rc_if.c
+++ b/src/rootcheck/check_rc_if.c
@@ -17,6 +17,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <pthread.h>
 
 #ifdef SOLARIS
 #include <stropts.h>

